backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/work/impl/LuceneWorkFactory.java

```diff
@@ -16,9 +17,10 @@
 
 	LuceneIndexWork<?> add(String indexName, String tenantId, String id, String routingKey, LuceneIndexEntry indexEntry);
 
-	LuceneIndexWork<?> update(String indexName, String tenantId, String id, String routingKey, LuceneIndexEntry indexEntry);
+	LuceneIndexWork<?> update(String indexName, MultiTenancyStrategy multiTenancyStrategy, String tenantId, String id, String routingKey,
+			LuceneIndexEntry indexEntry);
 
-	LuceneIndexWork<?> delete(String indexName, String tenantId, String id, String routingKey);
+	LuceneIndexWork<?> delete(String indexName, MultiTenancyStrategy multiTenancyStrategy, String tenantId, String id, String routingKey);
```

Comment:
I seriously think this should be a private field and initialized through the constructor. After all, callers have absolutely no control over which multi-tenancy strategy to use, this is all based on configuration.

Also, you could define two types of "delete works": one to delete by term, the other to delete by query.
Then you wouldn't need the multi tenancy strategy *inside* the work anymore, you would just use the strategy to create the correct work in the work factory.
You could even delegate to the multi-tenancy strategy to create the work, which would remove the need to expose a getter plus two methods, only one of which will ever work for a given implementation.

I'm mentioning this because it's exactly we handle works in the Elasticsearch backend, at least in Search 5. In 6 we don't currently have the full code with dialects and everything, but this turned out pretty well in Search 5 so we'll probably copy the code as-is when we merge.